fetcher/fetcher_test.go

```diff
@@ -0,0 +1,110 @@
+package fetcher
+
+import (
+	"archive/zip"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/dcos/dcos-diagnostics/dcos"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_NewReturnErrorWhenCantCreateZip(t *testing.T) {
+	_, err := New("not_existing_dir", nil, nil, nil, nil)
```

Comment:
So if New() were to create the channels and client, then this call would just be `New("not_existing_dir")`, which is a lot easier to read and write.

Reply:
but then we lose control if channels are buffered or not

Reply:
Ya, @mrnugget pointed that out above